src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs

```diff
@@ -0,0 +1,113 @@
+                var cancellationToken = completionContext.CancellationToken;
+
+                if (!completionContext.CompletionOptions.ShowNameSuggestions)
+                {
+                    return;
+                }
+
+                var context = (CSharpSyntaxContext)await completionContext.GetSyntaxContextWithExistingSpeculativeModelAsync(document, cancellationToken).ConfigureAwait(false);
+                if (context.IsInNonUserCode)
+                {
+                    return;
+                }
+
+                // Do not show name suggestions for unbound "async" or "yield" identifier.
+                // Most likely user is using it as keyword, so name suggestion will just interfere them
+                if (context.TargetToken.IsKindOrHasMatchingText(SyntaxKind.AsyncKeyword) ||
+                    context.TargetToken.IsKindOrHasMatchingText(SyntaxKind.YieldKeyword))
+                {
+                    if (context.SemanticModel.GetSymbolInfo(context.TargetToken).GetAnySymbol() is null)
+                    {
+                        return;
+                    }
+                }
+
+                var nameInfo = await NameDeclarationInfo.GetDeclarationInfoAsync(document, position, cancellationToken).ConfigureAwait(false);
+                using var _ = ArrayBuilder<(string name, Glyph glyph)>.GetInstance(out var result);
+
+                var sortValue = 0;
+                foreach (var recommender in Recommenders)
+                {
+                    var names = await recommender.ProvideRecommendedNamesAsync(completionContext, document, context, nameInfo, cancellationToken).ConfigureAwait(false);
+                    result.AddRange(names);
+                }
+
+                if (result.Count == 0)
+                    return;
+
+                foreach (var (name, glyph) in result)
+                {
+                    // We've produced items in the desired order, add a sort text to each item to prevent alphabetization
+                    completionContext.AddItem(CreateCompletionItem(name, glyph, sortValue.ToString("D8")));
+                    sortValue++;
+                }
+
+                completionContext.SuggestionModeItem = CommonCompletionItem.Create(
+                    CSharpFeaturesResources.Name, displayTextSuffix: "", CompletionItemRules.Default);
+            }
+            catch (Exception e) when (FatalError.ReportAndCatchUnlessCanceled(e, ErrorSeverity.General))
+            {
+                // nop
+            }
+        }
+
+        private static CompletionItem CreateCompletionItem(string name, Glyph glyph, string sortText)
+        {
+            return CommonCompletionItem.Create(
+                name,
+                displayTextSuffix: "",
+                CompletionItemRules.Default,
+                glyph: glyph,
+                sortText: sortText,
+                description: CSharpFeaturesResources.Suggested_name.ToSymbolDisplayParts());
+        }
+    }
+}
```

Comment:
is this all a move? did anything hange?

Reply:
Yes. The only change (except simply moving code around) is extracting code into methods.